### PR TITLE
bfix!: handle the expired token

### PR DESCRIPTION
**Pull request by aman-singh7**

### CODE REVIEW

File: lib/viewmodels/home_viewmodel.dart [hunk 1]
```diff
@@ -93,15 +93,17 @@ class HomeViewModel extends BaseModel {
 
   Future logoutAndClearData() async {
     _dialogService.showCustomProgressDialog(title: 'Logging You Out');
-    await _userApi.userLogout();
+    await _userApi.userLogout().catchError((err) {
+      print('error in logout: $err');
```
Comment (Nitish145) [inline lib/viewmodels/home_viewmodel.dart:97]: No need to catch error here.

Comment (Nitish145) [reply lib/viewmodels/home_viewmodel.dart:97]: We will add a better error handling later

**Review by Nitish145 — APPROVED** [review]

File: lib/viewmodels/home_viewmodel.dart (hunk 1 above)
Comment (aman-singh7) [reply lib/viewmodels/home_viewmodel.dart:97]: Without catching the exception app gets stuck at `Logging You out`.

Comment (Nitish145) [reply lib/viewmodels/home_viewmodel.dart:97]: Yeah, so let's use try catch block here.

**Pull request closed without merging**